### PR TITLE
[feat] added placementGroup to full flavors

### PR DESCRIPTION
**Pull request by prajwalvathreya**
**What this PR does / why we need it**: This PR adds the placement groups to `full` flavors by default when a cluster is created

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: templates/flavors/rke2/full/kustomization.yaml [hunk 1]
```diff
@@ -71,3 +72,19 @@ patches:
                   device: /dev/sdc
               mounts:
                 - ["/dev/sdc", "/var/lib/etcd_data"]
+  - target:
+      group: infrastructure.cluster.x-k8s.io
+      version: v1alpha2
+      kind: LinodeMachineTemplate
+    patch: |-
+      apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+      kind: LinodePlacementGroup
```
Comment (komer3) [inline templates/flavors/rke2/full/kustomization.yaml:81]: question: Is this suppose to be kind: LinodePlacementGroup? Wouldn't this be LinodeMachineTemplate?

File: templates/flavors/k3s/full-vpcless/kustomization.yaml [hunk 2]
```diff
@@ -178,3 +179,19 @@ patches:
     patch: |-
       - op: remove
         path: /spec/template/spec/firewallRef
+  - target:
+      group: infrastructure.cluster.x-k8s.io
+      version: v1alpha2
+      kind: LinodeMachineTemplate
+    patch: |-
+      apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+      kind: LinodePlacementGroup
```
Comment (komer3) [inline templates/flavors/k3s/full-vpcless/kustomization.yaml:188]: question: Is this suppose to be kind: LinodePlacementGroup? Wouldn't this be LinodeMachineTemplate?

File: templates/flavors/k3s/full/kustomization.yaml [hunk 3]
```diff
@@ -32,3 +33,19 @@ patches:
         labels:
           cluster-autoscaler: "true"
           etcd-backup: "true"
+  - target:
+      group: infrastructure.cluster.x-k8s.io
+      version: v1alpha2
+      kind: LinodeMachineTemplate
+    patch: |-
+      apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+      kind: LinodePlacementGroup
```
Comment (amold1) [inline templates/flavors/k3s/full/kustomization.yaml:42]: Same comment as above from Omer's

File: templates/addons/placementGroups/linodePlacementGroup.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: LinodePlacementGroup
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  region: ${LINODE_REGION}
```
Comment (amold1) [inline templates/addons/placementGroups/linodePlacementGroup.yaml:7]: can we add credentialsRef here

File: templates/flavors/k3s/full-vpcless/kustomization.yaml (hunk 2 above)
Comment (prajwalvathreya) [reply templates/flavors/k3s/full-vpcless/kustomization.yaml:188]: Yup, that should be LinodeMachineTemplate! Changing it

File: templates/addons/placementGroups/linodePlacementGroup.yaml (hunk 4 above)
Comment (eljohnson92) [inline templates/addons/placementGroups/linodePlacementGroup.yaml:7]: can we also add credential refs to this similar to VPC/Firewalls
```suggestion
  credentialsRef:
    name: ${CLUSTER_NAME}-credentials
  region: ${LINODE_REGION}
```

Comment (prajwalvathreya) [reply templates/addons/placementGroups/linodePlacementGroup.yaml:9]: Sure

Comment (prajwalvathreya) [reply templates/addons/placementGroups/linodePlacementGroup.yaml:9]: After making this updated when I apply the generated template, I get the following error

Error from server (BadRequest): error when creating "tst-full-8v.yaml": LinodePlacementGroup in version "v1alpha2" cannot be handled as a LinodePlacementGroup: strict decoding error: unknown field "spec.credentialsRef.region"

File: templates/flavors/k3s/full-vpcless/kustomization.yaml [hunk 5]
```diff
@@ -178,3 +179,20 @@ patches:
...
+      kind: LinodeMachineTemplate
+      metadata:
+        name: ${CLUSTER_NAME}-control-plane
+      spec:
+        template:
+          spec:
+            placementGroupRef:
+              apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+              kind: LinodePlacementGroup
+              name: ${CLUSTER_NAME}
```
Comment (komer3) [inline templates/flavors/k3s/full-vpcless/kustomization.yaml:198]: question: do we need to add this for worker machines too? or is that not the intention for placement groups?

Comment (prajwalvathreya) [reply templates/flavors/k3s/full-vpcless/kustomization.yaml:198]: Nope, it's only for the control-plane nodes. I've updated that in the latest commit

**Review by komer3 — APPROVED** [review]
LGTM!

**Pull request merged**